sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -706,7 +842,16 @@ case class TimeAdd(start: Expression, interval: Expression)
  */
 // scalastyle:off line.size.limit
 @ExpressionDescription(
-  usage = "_FUNC_(timestamp, string timezone) - Assumes given timestamp is UTC and converts to given timezone.")
+  usage = "_FUNC_(timestamp, timezone) - Assumes given timestamp is UTC and converts to given timezone.",
```

Comment:
I recently ran into this function in https://issues.apache.org/jira/browse/SPARK-17971?focusedCommentId=15583676&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15583676 and found its behavior pretty confusing. If you're willing, maybe a couple sentences here and in "to_unix_timestamp" could help clarify, because, timestamps actually have no timezone.

I think I'd describe this as:

Given a timestamp, which corresponds to a certain time of day in UTC, returns another timestamp that corresponds to the same time of day in the given timezone.